veracruz-client/src/veracruz_client.rs

```diff
@@ -227,32 +227,10 @@ impl VeracruzClient {
         })
     }
 
-    fn check_role_permission(&self, role: &VeracruzRole) -> Result<(), VeracruzClientError> {
-        match self
-            .policy
-            .identities()
-            .iter()
-            .find(|&x| *x.certificate() == self.client_cert)
-        {
-            Some(identity) => match identity.roles().iter().find(|&x| *x == *role) {
-                Some(_) => Ok(()),
-                None => Err(VeracruzClientError::InvalidRoleError(
-                    self.client_cert.clone().into_bytes(),
-                    role.clone(),
-                )),
-            },
-            None => Err(VeracruzClientError::InvalidClientCertificateError(
-                self.client_cert.to_string(),
-            )),
-        }
-    }
-
-    pub fn send_program(&mut self, program: &Vec<u8>) -> Result<(), VeracruzClientError> {
-        self.check_role_permission(&VeracruzRole::ProgramProvider)?;
-
+    pub fn send_program(&mut self, file_name:&str, program: &Vec<u8>) -> Result<(), VeracruzClientError> {
```

Comment:
By complete accident I found out that `send_program` and `send_data` are interchangeable now. Could these be deduplicated (maybe `send_file`)?

Reply:
My current thinking is that the CLIs (https://github.com/veracruz-project/veracruz/pull/75) need to wait for both the renames and this and the WASI changes. But that also means this and other API changes could go into the CLIs PR if needed.